src/System.Linq/tests/ShortCircuitingTests.cs

```diff
@@ -75,7 +81,10 @@ void MinNullableDoubleDoesntCheckAll()
             var tracker = new TrackingEnumerable(10);
             var source = tracker.Select(i => (double?)(i == 5 ? double.NaN : (double)i));
             Assert.True(double.IsNaN(source.Min().GetValueOrDefault()));
-            Assert.Equal(5, tracker.Moves);
+
+            // .NET core shortcircuits as an optimization.
+            // See https://github.com/dotnet/corefx/pull/2350.
+            Assert.Equal(PlatformDetection.IsFullFramework ? 10 : 5, tracker.Moves);
```

Comment:
@VSadov, are we ok with this behavioral difference? (I'll stop asking about these.)